client/db/src/lib.rs

```diff
@@ -515,12 +515,18 @@ impl<Block: BlockT> sc_client_api::blockchain::Backend<Block> for BlockchainDb<B
 						)),
 					},
 					TransactionStorageMode::StorageChain => {
-						match Vec::<Block::Hash>::decode(&mut &body[..]) {
-							Ok(hashes) => {
-								let extrinsics: ClientResult<Vec<Block::Extrinsic>> = hashes.into_iter().map(
-									|h| self.extrinsic(&h).and_then(|maybe_ex| maybe_ex.ok_or_else(
-										|| sp_blockchain::Error::Backend(
-											format!("Missing transaction: {}", h))))
+						match Vec::<(Block::Hash, Vec<u8>)>::decode(&mut &body[..]) {
```

Comment:
This body structure is really implicitly used here. Maybe we could use at least some typedef.